File: Problemset/1753-Maximum Score From Removing Stones/Mathematical.rb
# @param {Integer} a
# @param {Integer} b
# @param {Integer} c
# @return {Integer}
def maximum_score(a, b, c)
  m = [a, b, c].max
  s = [a, b, c].sum

  s <= 2 * m ? s - m : s / 2
end

File: Problemset/1753-Maximum Score From Removing Stones/Mathematical.rs
impl Solution {
    pub fn maximum_score(a: i32, b: i32, c: i32) -> i32 {
        let m = a.max(b).max(c);
        let s = a + b + c;

        if s <= 2 * m {
            s - m
        } else {
            s / 2
        }
    }
}
